File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        int a = n;
        int t = 100;

        while (t > 0) {
            int b = 0;
            while (a > 0) {
                int d = a % 10;
                b += d * d;
                a /= 10;
            }

            if (b == 1) {
                return true;
            }

            a = b;
            t--;
        }
        
        return false;
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        a = n
        t = 100
        while (t):
            b = 0
            for i in str(a):
                b += int(i)**2
                
            if b==1:
                return True
            a = b
            t-=1
        return False